tests/test_upload.py

```diff
@@ -125,6 +125,81 @@ def test_deprecated_repo(tmpdir):
     )
 
 
+def test_upload_prints_skip_message_for_uploaded_package(tmpdir, capsys):
+    pypirc = os.path.join(str(tmpdir), ".pypirc")
+    dists = ["tests/fixtures/twine-1.5.0-py2.py3-none-any.whl"]
+
+    with open(pypirc, "w") as fp:
+        fp.write(textwrap.dedent("""
+            [pypi]
+            username:foo
+            password:bar
+        """))
+
+    upload_settings = settings.Settings(
+        repository_name="pypi", sign=None, identity=None, username=None,
```

Comment:
Style nit: when argument lists end up taking up multiple lines, prefer to make each argument live on its own line:

```
upload_settings = settings.Settings(
    repository_name="pypi",
    sign=None,
    identity=None,
    ...
)
```

Also, since most of these are just `None`, can you omit them?